# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/base/RangeUtils.cpp
# Commit: 3aa74b745aa5
# Full Hash: 3aa74b745aa5ae1f1b460452f223ad0b9633fec0
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-05-13 09:26:34
# Description:
#   Bug 1965664 - Fix some crashes in ContentSubtreeIterator r=jjaschke,dom-core
#   
#   by skipping the shadow hosts that ShadowDOM selection doesn't
#   support at the moment.
#   
# ==============================================================================

diff -r ecccb8942de4 -r 3aa74b745aa5 dom/base/RangeUtils.cpp
--- a/dom/base/RangeUtils.cpp	Mon May 12 20:34:06 2025 +0000
+++ b/dom/base/RangeUtils.cpp	Mon May 12 20:40:53 2025 +0000
@@ -365,7 +365,9 @@
   if (StaticPrefs::dom_shadowdom_selection_across_boundary_enabled() &&
       aAllowCrossShadowBoundary == AllowRangeCrossShadowBoundary::Yes) {
     if (aNode.IsContent()) {
-      if (HTMLSlotElement* slot = aNode.AsContent()->GetAssignedSlot()) {
+      if (HTMLSlotElement* slot = aNode.AsContent()->GetAssignedSlot();
+          slot && GetShadowRoot(slot->GetContainingShadowHost(),
+                                aAllowCrossShadowBoundary)) {
         return slot;
       }
     }